src/components/button/button.svelte

```diff
@@ -79,7 +80,22 @@
   on:click={onClick}
   {...$$restProps}
 >
-  <slot>Leo Button</slot>
+  {#if isLoading}
+    {#if $$slots.loading}
+      <slot name="loading" />
+    {:else if !fab}
+      <div>
+        <slot>Leo Button</slot>
+      </div>
+    {/if}
+    <ProgressRing />
```

Comment:
> I feel like this should maybe live inside the `loading` slot as the default content, so people can override it?

I don't believe that the design is supposed to be overridden in this case. cc @aguscruiz

> Do we actually need the loading slot? People can just change the button content if it's loading:

Arguably yes, and people could still do this if they prefer, but IMO it's cleaner for the consumer to not clutter the button contents up with logic, at least when that logic is going to be a super common use case of loading.

Reply:
Okay, I don't have a strong preference here, it just feels slightly off to me (probably because of my React background).

```tsx
// so we're going to use the button like this:
<Button>
    <span slot="loading">Loading...</span>
   Click Me
</Button>

// instead of
<Button>{loading ? "Loading..." : "Click Me"}</Button>
```

I guess there's still nothing stopping you using it the more Reacty way anyway.

Reply:
@petemill, I think you may have mentioned something about this on our call last week. Any strong preferences?

Reply:
Well on the one hand it's kind of like `placeholder` text - you don't conditionally put it in when it's empty.

But on the other hand it's _not_ like that because the control itself isn't determining when the button is in a loading state - the consumer is.

At least the consumer can do it either way. Both these should work:

```svelte
<Button isLoading={thingInProgress}>
    <span slot="loading">Loading...</span>
   Click Me
</Button>
```

```svelte
<Button isLoading={thingInProgress}>
  {thingInProgress ? "Loading..." : "Click Me"}
</Button>
```

It's kind of the same with the new icon placeholders - the button is going to decide to show / hide them, and not the consumer. But the consumer can also put icons in the default slot and decide when to show / hide them. I guess it's fine to have the option to override?